src/trace/span-inferrer.spec.ts

```diff
@@ -53,6 +57,107 @@ describe("SpanInferrer", () => {
     });
   });
 
+  it("remaps all SNS inferred span service name based on DD_SERVICE_MAPPING", () => {
+    // Set the environment variable that depicts test outcome, i.e. the result of service tag
+    process.env.DD_SERVICE_MAPPING = "lambda_sns:new-name";
+    const inferrer = new SpanInferrer(mockWrapper as unknown as TracerWrapper);
+    inferrer.createInferredSpan(snsEvent, {} as any, {} as SpanContext);
+
+    let modifiedSnsEvent = JSON.parse(JSON.stringify(snsEvent));
+    modifiedSnsEvent.Records[0].EventSubscriptionArn = "arn:aws:sns:us-east-1:123456789012:DifferentTopic";
```

Comment:
instead of creating a whole new event we can just edit some characteristics to simulate a different even